csrc/python_frontend/fusion_definition.cpp

```diff
@@ -105,6 +105,9 @@ void FusionDefinition::finalizeDefinition() {
     trie_node_ = child_node.value();
     fusion_id_ = std::optional<size_t>(trie_node_->fusion_id);
   }
+
+  NVF_ERROR(num_recording_states_presched_ == 0);
```

Comment:
nitpick, std::nullopt might read easier here. Also an error message would be nice.

Reply:
I added an error message.

Reply:
It is a part of FusionDefinition state, so I don't see it as `std::optional`.